htlcswitch/hop/iterator.go

```diff
@@ -315,6 +308,27 @@ func deriveBlindedRouteForwardingInfo(r *sphinxHopIterator,
 		return nil, routeRole, err
 	}
 
+	nextHopIsDummy := checkForDummyHop(routeData, r.router.OnionPublicKey())
+	if err != nil {
```

Comment:
This func doesn't return an error, so it's checking the error (returned alongside `nextEph`) again.

Reply:
good catch 🙏